src/BuiltInTools/dotnet-watch/Filters/BrowserRefreshServer.cs

```diff
@@ -61,23 +68,26 @@ public async ValueTask<string> StartAsync(CancellationToken cancellationToken)
 
             await _refreshServer.StartAsync(cancellationToken);
 
-            var serverUrl = _refreshServer.Services
+            var serverUrls = _refreshServer.Services
                 .GetRequiredService<IServer>()
                 .Features
                 .Get<IServerAddressesFeature>()
-                .Addresses
-                .First();
+                .Addresses;
 
             if (envHostName is null)
             {
-                return useTls ?
-                    serverUrl.Replace("https://127.0.0.1", "wss://localhost", StringComparison.Ordinal) :
-                    serverUrl.Replace("http://127.0.0.1", "ws://localhost", StringComparison.Ordinal);
+                return serverUrls.Select(s =>
+                    s.Replace("http://127.0.0.1", "ws://localhost", StringComparison.Ordinal)
+                    .Replace("https://127.0.0.1", "wss://localhost", StringComparison.Ordinal));
             }
 
-            return serverUrl
-                .Replace("https://", "wss://", StringComparison.Ordinal)
-                .Replace("http://", "ws://", StringComparison.Ordinal);
+            return new[]
+            {
+                serverUrls
+                    .First()
```

Comment:
Why do we only pick the first URL here?

Reply:
Users can only provide one URL via an environment variable. We previously had this .First https://github.com/dotnet/sdk/pull/18039/files#diff-2696395053559822320fce8c67b5b71b00f693255b29906d02c6427be3e9533fL69 here.